### PR TITLE
Add yaml validation using jsonschema.

### PR DESCRIPTION
**Pull request by robertwb**
------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

File: sdks/python/apache_beam/yaml/yaml_transform.py [hunk 1]
```diff
@@ -36,6 +37,24 @@
 _LOGGER = logging.getLogger(__name__)
 yaml_provider.fix_pycallable()
 
+try:
+  import jsonschema
+except ImportError:
+  jsonschema = None
+
+if jsonschema is not None:
+  with open(os.path.join(os.path.dirname(__file__),
```
Comment (jrmccluskey) [inline sdks/python/apache_beam/yaml/yaml_transform.py:46]: The PythonDocs PreCommit does not like the pipeline.schema.yaml file read here (https://ci-beam.apache.org/job/beam_PreCommit_PythonDocs_Commit/13751/console)

File: sdks/python/apache_beam/yaml/yaml_transform.py [hunk 2]
```diff
@@ -19,6 +19,7 @@
 import json
 import logging
 import pprint
+import os
```
Comment (jrmccluskey) [inline sdks/python/apache_beam/yaml/yaml_transform.py:22]:
```suggestion
import logging
import os
import pprint
```

File: sdks/python/apache_beam/yaml/yaml_transform.py [hunk 3]
```diff
@@ -627,6 +646,7 @@ class YamlTransform(beam.PTransform):
   def __init__(self, spec, providers={}):  # pylint: disable=dangerous-default-value
     if isinstance(spec, str):
       spec = yaml.load(spec, Loader=SafeLineLoader)
+    # TODO: Validate as a transform.
```
Comment (jrmccluskey) [inline sdks/python/apache_beam/yaml/yaml_transform.py:649]: Can we tie this ToDo to a GitHub Issue for tracking?

File: sdks/python/apache_beam/yaml/yaml_transform.py (hunk 1 above)
Comment (robertwb) [reply sdks/python/apache_beam/yaml/yaml_transform.py:46]: Added as a data file.

File: sdks/python/apache_beam/yaml/yaml_transform.py (hunk 2 above)
Comment (robertwb) [reply sdks/python/apache_beam/yaml/yaml_transform.py]: Done.

**Review by jrmccluskey — APPROVED** [review]
LGTM, just holding off on merging until the last unit test suites pass

**Pull request merged**